src/transformers/quantizers/quantizer_torchao.py

```diff
@@ -73,6 +73,50 @@ def validate_environment(self, *args, **kwargs):
                     )
                 else:
                     self.offload = True
+        if self.pre_quantized:
+            safe_globals = []
```

Comment:
if we do `import torchao`, I think we should get everything here? otherwise we'd need to fix torchao

Reply:
I'm using torchao 0.5.0 and it's not working on my side. I can try with the latest tomorrow !

Reply:
I see, it's not expected I think, I think it should be fixed in torchao side, I feel 0.5 should have this functionality already actually. if you can have a standalone repro that will be very helpful for us. I remember I have tested in https://huggingface.co/docs/transformers/main/en/quantization/torchao

Reply:
Actually, we ran into this issue with @MekkCyber on the example you shared in the docs. 
Here's a the reproducer, let us know if you also have this issue : 

```python
from transformers import TorchAoConfig, AutoTokenizer, AutoModelForCausalLM
import torch

model_name = "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
quant_config = TorchAoConfig("int4_weight_only", group_size=32)
quantized_model = AutoModelForCausalLM.from_pretrained(
    model_name,
    torch_dtype=torch.bfloat16,
    device_map="cuda:0",
    quantization_config=quant_config,
)
output_dir = "llama3-8b-int4wo-128"
quantized_model.save_pretrained(output_dir, safe_serialization=False)

loaded_quantized_model = AutoModelForCausalLM.from_pretrained(output_dir, device_map="cuda")
```

Reply:
OK will test and report back

Reply:
It works for torchao >= 0.5.0 and torch >= 2.5.1

Reply:
@SunMarc can you help double check? what is the torchao and pytorch version you are using

Reply:
ok this might be an issue with the torch version. I tested with 2.5 and it works but fails with 2.4. Can you try with this version ?  If we can't fix it for older version of torch, we can either do a check for torch>=2.5 or merge this PR.

Reply:
yeah we can't really fix older versions since these are already released I think, I remember there was some missing features in pytorch in 2.4 as well, I feel it's fine to just enable torchao for 2.5 and later

Reply:
Sounds good ! I pinned torch version when using prequnatized model